AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Storage/SQLite/Statement+Model.swift

```diff
@@ -121,13 +121,24 @@ extension Statement: StatementModelConvertible {
             return convertCollection(field: field, schema: schema, from: element, path: path)
 
         case (.model, false):
-            let foreignKey = field.association.map(getTargetNames).map {
-                field.foreignKeySqlName(withAssociationTargets: $0)
-            }
-            let targetBinding = foreignKey.flatMap {
+            let targetNames = getTargetNames(field: field)
+            var associatedFieldValues = targetNames.map {
                 getValue(from: element, by: path + [$0])
+            }.compactMap { $0 }
+            .map { String(describing: $0) }
+
+            if associatedFieldValues.isEmpty {
+                associatedFieldValues = associatedValues(
+                    from: path + [field.foreignKeySqlName(withAssociationTargets: targetNames)],
+                    element: element
+                )
+            }
+            guard associatedFieldValues.count == field.associatedFieldNames.count else {
+                return nil
```

Comment:
I took a stab at the problem we're solving by remove the required codegen changes from this PR, over in PR https://github.com/aws-amplify/amplify-swift/pull/2749/
- removed the public API changes that depended on codegen to generate the new `associatedFields` for hasOne and belongsTo. 
- Removed retrieving the associatedFieldNames at this line of code, basically just padded it with empty strings (see [code](https://github.com/aws-amplify/amplify-swift/pull/2749/files#r1098937871))